app/src/main/java/com/example/trainingintive/di/NetworkModule.kt

```diff
@@ -33,9 +34,25 @@ class NetworkModule {
             .build()
     }
 
+    @Provides
+    @Singleton
+    @Named("dogImageRetrofit")
```

Comment:
To operate on hard typed reference you can use dedicated annotation
https://medium.com/mindorks/qualifiers-in-dagger-android-development-c0574cca87a2